src/client.rs

```diff
@@ -84,19 +109,12 @@ impl<C> Client<C> {
 
 impl<C: HttpClient> Client<C> {
     /// Makes a request to a Matrix API endpoint.
-    pub async fn send_request<R: OutgoingRequest>(&self, request: R) -> ResponseResult<C, R> {
-        self.send_customized_request(request, |_| Ok(())).await
-    }
-
-    /// Makes a request to a Matrix API endpoint including additional URL parameters.
-    pub async fn send_customized_request<R, F>(
-        &self,
-        request: R,
-        customize: F,
-    ) -> ResponseResult<C, R>
+    pub async fn send_request<R>(&self, request: R) -> ResponseResult<C, R>
     where
+        C::RequestBody: AsRef<[u8]>,
```

Comment:
This bound should be added directly to the definition of `RequestBody`, instead of adding it to each function.